spark-plugins/src/main/java/co/cask/hydrator/plugin/spark/KinesisStreamingSource.java

```diff
@@ -0,0 +1,314 @@
+
+    @Name("streamName")
+    @Description("The name of the Kinesis stream to the get the data from. The stream should be active")
+    @Macro
+    private final String streamName;
+
+    @Name("endpointUrl")
+    @Description("Valid Kinesis endpoints URL eg. kinesis.us-east-1.amazonaws.com")
+    @Macro
+    private final String endpoint;
+
+    @Name("region")
+    @Description("AWS region specific to the stream")
+    @Macro
+    private final String region;
+
+    @Name("duration")
+    @Description("The interval (e.g., Duration(2000) = 2 seconds) at which the Kinesis Client Library saves its " +
+      "position in the stream.")
+    @Macro
+    private final Integer duration;
+
+    @Name("initialPosition")
+    @Description("Can be either TRIM_HORIZON or LATEST, Default position will be Latest")
+    private final String initialPosition;
+
+    @Name("awsAccessKeyId")
+    @Description("AWS access Id having access to Kinesis streams")
+    @Macro
+    private final String awsAccessKeyId;
+
+    @Name("awsAccessSecret")
+    @Description("AWS access key secret having access to Kinesis streams")
+    @Macro
+    private final String awsAccessSecret;
+
+    @Description("Optional format of the Kinesis stream shard. Any format supported by CDAP is supported. For " +
+      "example, a value of 'csv' will attempt to parse Kinesis payloads as comma-separated values. If no format is " +
+      "given, Kinesis shard payloads will be treated as bytes.")
+    @Nullable
+    private final String format;
+
+    @Description("Output schema of the source, The fields are used in conjunction with the format to parse Kinesis " +
+      "payloads")
+    private final String schema;
+
+    public KinesisStreamConfig(String referenceName, String appName, String streamName, String endpoint, String region,
+                               Integer duration, String initialPosition, String awsAccessKeyId, String awsAccessSecret,
+                               String format, String schema) {
+      super(referenceName);
+      this.appName = appName;
+      this.streamName = streamName;
+      this.endpoint = endpoint;
+      this.region = region;
+      this.duration = duration;
+      this.initialPosition = initialPosition;
+      this.awsAccessKeyId = awsAccessKeyId;
+      this.awsAccessSecret = awsAccessSecret;
+      this.format = format;
+      this.schema = schema;
+    }
+
+    @Nullable
+    public String getFormat() {
+      return Strings.isNullOrEmpty(format) ? null : format;
+    }
+
+    public Schema parseSchema() {
+      try {
+        return Schema.parseJson(schema);
+      } catch (IOException | NullPointerException e) {
+        throw new IllegalArgumentException("Invalid schema: " + e.getMessage());
+      }
+    }
+
+    public void validate() {
+      Schema messageSchema = parseSchema();
+      // if format is empty, there must be just a single message field of type bytes or nullable types.
+      if (Strings.isNullOrEmpty(format)) {
```

Comment:
This whole code block till line 303 is duplicated in KafkaConfig. Will be better to refactor it